sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageInputStream.java

```diff
@@ -286,7 +286,7 @@ public int read(final byte[] b, final int off, final int len) throws IOException
         int chunks = (int) (Math.ceil((double) len / (double) this.chunkSize));
         int numOfBytesRead = 0;
         for (int i = 0; i < chunks; i++) {
-            int results = this.readInternal(b, off + numOfBytesRead, len);
+            int results = this.readInternal(b, off + numOfBytesRead, len - numOfBytesRead);
```

Comment:
nice catch!

Reply:
may not be relevant to this PR, but do we want to add tests for this fix?

Reply:
I added one to NioBlobInputStream. It looks like that may get moved into regular BlobInputStream, though.